### PR TITLE
DOC: Added that threads can now contain a metadata stream

### PR DESCRIPTION
**Pull request by j-t-1**
Also some very small changes

### CODE REVIEW

File: docs/user/extract-images.md [hunk 1]
```diff
@@ -1,8 +1,5 @@
 # Extract Images
 
-> Please note: In order to use the following code you need to install optional
```
Comment (stefan6419846) [inline docs/user/extract-images.md:3]: Why remove this? This is still relevant as *Pillow* is an optional dependency.

File: docs/user/merging-pdfs.md [hunk 2]
```diff
@@ -77,13 +77,14 @@ It is recommended to use `append` or `merge` instead.
 
 ## Merging forms
 
-When merging forms, some form fields may have the same names, preventing access to some data.
+When merging forms, some form fields may have the same names, preventing access
```
Comment (stefan6419846) [inline docs/user/merging-pdfs.md:80]: I consider this an unnecessary change.

File: docs/user/merging-pdfs.md [hunk 3]
```diff
@@ -92,10 +93,11 @@ fields will be listed.
 
 ## reset_translation
 
-During cloning, if an object has been already cloned, it will not be cloned again, and a pointer
-to this previously cloned object is returned instead. Because of that, if you add/merge a page that has
-already been added, the same object will be added the second time. If you modify any of these two pages later,
-both pages can be modified independently.
+During cloning, if an object has been already cloned, it will not be cloned
```
Comment (stefan6419846) [inline docs/user/merging-pdfs.md:96]: Do we really have any benefits from this?

File: docs/user/merging-pdfs.md [hunk 4]
```diff
@@ -122,11 +124,12 @@ The same holds true if you try to clone an object twice. It will return the prev
 assert object.clone(writer) == object.clone(writer)
 ```
 
-Please note that if you clone an object, you will clone all the objects below as well,
-including the objects pointed by *IndirectObject*. Due to this, if you clone a page that
-includes some articles (`"/B"`), not only the first article, but also all the chained articles
-and the pages where those articles can be read will be copied.
-This means that you may copy lots of objects which will be saved in the output PDF as well.
+Please note that if you clone an object, you will clone all the objects below as
```
Comment (stefan6419846) [inline docs/user/merging-pdfs.md:127]: Do we really have any benefits from this?

File: docs/user/viewer-preferences.md [hunk 5]
```diff
@@ -1,15 +1,16 @@
 # Adding Viewer Preferences
 
-It is possible to set viewer preferences of a PDF file.
-These properties are described in Section 12.2 of the [PDF 1.7 specification](https://opensource.adobe.com/dc-acrobat-sdk-docs/pdfstandards/PDF32000_2008.pdf).
+It is possible to set viewer preferences of a PDF file.  These properties are
```
Comment (stefan6419846) [inline docs/user/viewer-preferences.md:3]: Do we really have any benefits from this? Additionally, there is a whitespace too much.

File: docs/user/viewer-preferences.md [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
+described in Section 12.2 of the [PDF 1.7
+specification](https://opensource.adobe.com/dc-acrobat-sdk-docs/pdfstandards/PDF32000_2008.pdf).
 
-Note that the `/ViewerPreferences` dictionary does not exist by default.
-If it is not already present, it must be created by calling the `create_viewer_preferences` method
-of the `PdfWriter` object.
+Note that the `/ViewerPreferences` dictionary does not exist by default.  If it
```
Comment (stefan6419846) [inline docs/user/viewer-preferences.md:7]: See above.

File: docs/user/viewer-preferences.md [hunk 7]
```diff
@@ -1,15 +1,16 @@
...
+is not already present, it must be created by calling the
+`create_viewer_preferences` method of a `PdfWriter` object.
 
-If viewer preferences exist in a PDF file being read with `PdfReader`,
-you can access them as properties of `viewer_preferences`.
-Otherwise, the `viewer_preferences` property will be set to `None`.
+If viewer preferences exist in a PDF file being read with `PdfReader`, you can
```
Comment (stefan6419846) [inline docs/user/viewer-preferences.md:11]: See above.

File: docs/user/extract-images.md (hunk 1 above)
Comment (j-t-1) [reply docs/user/extract-images.md:3]: The example does not use Pillow, but some of the other examples do.

File: pypdf/_doc_common.py [hunk 8]
```diff
@@ -847,8 +847,14 @@ def threads(self) -> Optional[ArrayObject]:
...
 
-        It is an array of dictionaries with "/F" and "/I" properties or
+        It is an array of dictionaries with "/F" (the first bead in the thread)
+        and "/I" (a thread information dictionary containing information about
+        the thread, such as its title, author, and creation date) properties or
         None if there are no articles.
+
+        Since PDF 2.0 it can also contain an indirect reference to a metadata
+        stream containing information about the thread, such as its title,
+        author, and creation date.
```
Comment (stefan6419846) [inline pypdf/_doc_common.py:857]:
```suggestion
        author and creation date.
```

File: docs/user/merging-pdfs.md (hunk 2 above)
Comment (j-t-1) [reply docs/user/merging-pdfs.md:80]: Agree,

File: pypdf/_doc_common.py [hunk 9]
```diff
@@ -998,16 +1004,13 @@ def _build_outline_item(self, node: DictionaryObject) -> Optional[Destination]:
...
-        Provides the capability to remove a page/range of page from the list
-        (using the del operator).
-        Note: only the page entry is removed. As the objects beneath can be used
-        elsewhere.
-        A solution to completely remove them - if they are not used anywhere -
-        is to write to a buffer/temporary file and to load it into a new PdfWriter
-        object afterwards.
+        This property allows to get a page or a range of pages.
+
+        For PdfWriter only: provides the capability to remove a page/range of
```
Comment (stefan6419846) [inline pypdf/_doc_common.py:1009]: * Please start with an uppercase letter after the colon.
* Please avoid two whitespace characters between words/sentences.
* Please use Sphinx-syntax to format the note accordingly.

File: docs/user/extract-images.md (hunk 1 above)
Comment (stefan6419846) [reply docs/user/extract-images.md:3]: But AFAIK image extraction will throw an *ImportError* when *Pillow* is not available.

File: docs/user/viewer-preferences.md (hunk 7 above)
Comment (j-t-1) [reply docs/user/viewer-preferences.md:11]: Will change this.

File: docs/user/viewer-preferences.md (hunk 6 above)
Comment (j-t-1) [reply docs/user/viewer-preferences.md:7]: Will change this.

File: pypdf/_doc_common.py (hunk 8 above)
Comment (j-t-1) [reply pypdf/_doc_common.py:857]: > A metadata stream containing information about the thread, such as its title, author, and creation date
I copied this from the specification.

File: pypdf/_doc_common.py (hunk 9 above)
Comment (j-t-1) [reply pypdf/_doc_common.py:1009]: Will do. I will change the "Note: " to "Remember:" to avoid having a note within a note.

File: docs/user/viewer-preferences.md (hunk 5 above)
Comment (j-t-1) [reply docs/user/viewer-preferences.md:3]: Will change this.

File: docs/user/merging-pdfs.md (hunk 3 above)
Comment (j-t-1) [reply docs/user/merging-pdfs.md:96]: It only changes the source file (slightly better?) not the documentation webpage.

File: docs/user/merging-pdfs.md (hunk 4 above)
Comment (j-t-1) [reply docs/user/merging-pdfs.md:127]: Not really, but I will delete a superfluous space.

File: docs/user/merging-pdfs.md (hunk 3 above)
Comment (stefan6419846) [reply docs/user/merging-pdfs.md:95]: I think we should avoid modifying the line length of source files without having any specific reason to reduce the general noise.

File: pypdf/_doc_common.py [hunk 10]
```diff
@@ -998,16 +1004,15 @@ def _build_outline_item(self, node: DictionaryObject) -> Optional[Destination]:
...
-        Provides the capability to remove a page/range of page from the list
-        (using the del operator).
-        Note: only the page entry is removed. As the objects beneath can be used
-        elsewhere.
-        A solution to completely remove them - if they are not used anywhere -
-        is to write to a buffer/temporary file and to load it into a new PdfWriter
-        object afterwards.
+        This property allows to get a page or a range of pages.
+
+        .. note::
```
Comment (stefan6419846) [inline pypdf/_doc_common.py:1009]: This is formatted incorrectly and thus ignored: https://pypdf--2711.org.readthedocs.build/en/2711/modules/PdfWriter.html#pypdf.PdfWriter.pages I recommend you to test the correct rendering locally. The last time we did this looked like this: https://github.com/py-pdf/pypdf/pull/2695/files#diff-2376a2cb969cf3bd675e8f2a9623f3379907210fc6532ffd1e84b9cadae9b37dR98-R100

Comment (j-t-1) [reply pypdf/_doc_common.py]: Changed.

Comment (stefan6419846) [reply pypdf/_doc_common.py]: This still looks broken. I am not sure what the correct command is here, but apparently now we have some code rendering.

The alternative approach to try here might be something like this:

```
.. note::
   The text
```

Comment (j-t-1) [reply pypdf/_doc_common.py]: Will read about how this works and work out best way.

File: docs/user/extract-images.md (hunk 1 above)
Comment (j-t-1) [reply docs/user/extract-images.md]: Reinstated. The file should be the same as what is in the current HEAD.

File: docs/user/merging-pdfs.md (hunk 4 above)
Comment (j-t-1) [reply docs/user/merging-pdfs.md:125]: Done.

File: docs/user/viewer-preferences.md (hunk 7 above)
Comment (j-t-1) [reply docs/user/viewer-preferences.md:10]: Done.

**Review by stefan6419846 — APPROVED** [review]
